hcipy/optics/aberration.py

```diff
@@ -51,6 +52,49 @@ def psd(grid):
 
     return Field(screen * aperture, pupil_grid)
 
+def make_high_pass_power_law_error(pupil_grid, std, diameter, cutoff_frequency, exponent=-2.5, aperture=None, filter_shape_parameter=15):
+    '''Create an error surface from a high-pass filtered power-law power spectral density.
+
+    Parameters
+    ----------
+    pupil_grid : Grid
+        The grid on which to calculate the error.
+    std : scalar
+        The standard deviation of the wavefront aberration in meters.
+    diameter : scalar
+        The diameter over which the ptv is calculated.
+    cutoff_frequency : scalar
+        The cutoff frequency of the high-pass filter.
+    exponent : scalar
+        The exponent of the power law.
+    aperture : Field
+        The mask over which to calculate the ptv. A circular aperture with diameter
+        `diameter` is used if this is not given.
+    filter_shape_parameter : scalar
+        The shape parameter for the general Gaussian high-pass filter fuction. Default value is 15.0
+
+    Returns
+    -------
+    Field
+        The surface error calculated on `pupil_grid`.
+    '''
+    def filter_function(fourier_grid):
+        return 1 - make_general_gaussian_aperture(2 * cutoff_frequency, filter_shape_parameter)(fourier_grid)
+
+    ff = FourierFilter(pupil_grid, filter_function, q=2)
+
+    if aperture is None:
+        aperture = make_circular_aperture(diameter)(pupil_grid)
+    aperture_mask = aperture > 0
+
+    phase_screen = make_power_law_error(pupil_grid, 1.0, diameter, exponent=exponent, aperture=None)
+    phase_screen = np.real(ff.forward(phase_screen + 0j))
```

Comment:
You need the `+ 0j`?

Reply:
Adding the `+ 0j` creates an implicit copy of the array. Some FFT backends have specialized functions for a real FFT with a full complex output without an intermediate copy. So by adding 0j, you'd be actively ignoring that potential optimization.

I'm surprised by the warning. That's obviously the case for complex->real casts without taking the real part (which you're doing here). But there shouldn't be a warning for real->complex casts, which are implicit.